### ISSUE TITLE
Review restructuring routerTypes "navigation" and router.

### ISSUE THREAD

**Issue by cdcabrera** [RedHatInsights/curiosity-frontend#290]
## Description
As a developer removing the platform method for "setNavigation" left an awkward implementation around product id look ups and routing, given by the navigation type under "routerTypes", and the associated "routerHelpers". This data is passed to the views (any component tied into the router).

## Acceptance Criteria
- [x] RouterTypes have been researched, platform methods have been confirmed on the ~frontend components~ chroming repo
- [x] ~RouterTypes have been consolidated between the "sub" navigation and "sub" router types~ see #375 

## Assumptions and Questions
- Product IDs are funneled through RouterTypes. We may be able to streamline/remove the navigation structure type to simply be a product ID lookup instead, still giving us the ability to submit the Product IDs to the RHSM API... typically passed through a prop within the views.

**Comment by zanewoodfin** [comment]
It's very difficult at this time to remove the navigation routerType.  I tried to do so and quickly found myself with a large amount of code changes for questionable success.  The navigation routerType is currently needed for 2 different mappings:
- id to path: used when the event is captured for a nav link click to generate the "routeHref" for loading a new page
- id to pathParameter: used for the productId and key when rendering charts

To resolve the routeHref usage, we would need chrome to be updated to provide us with the actual path when a nav link is clicked instead of just the "navId" as they call it.

For rhel-sw, "pathParameter" could be determined from the variant url parameter if we merged the `/rhel-sw/all` and `rhel-sw/:variant` routes.  To merge the two routes we would need to figure out a new way to handle the default redirect to `rhel-sw/all`, and we would still need some sort of mapping, albeit a simpler one.  For openshift-sw, "pathParameter" could just be explicitly stated in the view because there is only one openshift-sw variant.
